File: frontend/src/app/sites/nobody-is-perfect/dto/GameRoomState.ts
import {GameRoom} from "./GameRoom";
import {GameRoomStateBase} from "../../core/ws/dto/GameRoomStateBase";
import {GameRoomMember} from "../../core/ws/dto/GameRoomMember";
import {Prompt} from "./Prompt";
import {Round} from "./Round";

export interface GameRoomState extends GameRoomStateBase {
  room: GameRoom;
  gameMaster: GameRoomMember;
  rounds: Round[];
}

File: backend/src/main/java/net/zorphy/backend/site/airsoft/dto/GameRoomState.java
package net.zorphy.backend.site.airsoft.dto;

import net.zorphy.backend.site.core.ws.dto.GameRoom;
import net.zorphy.backend.site.core.ws.dto.GameRoomPrivateStateBase;
import net.zorphy.backend.site.core.ws.dto.GameRoomStateBase;

public record GameRoomState(
        GameRoom room
) implements GameRoomStateBase {
    @Override
    public GameRoomStateBase toPublicState() {
        return this;
    }

    @Override
    public GameRoomPrivateStateBase toPrivateState(String username) {
        return new GameRoomPrivateState();
    }
}
